share/wake/lib/core/examples.wake

```diff
@@ -0,0 +1,191 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+package wake
+
+def expectOne (msg: String) (result: Result (List a) Error): Result a Error = match result
+    Fail err      = addErrorContext msg (Fail err)
+    Pass (x, Nil)  = Pass x
+    Pass otherwise = failWithError "{msg}: expected 1, got {len otherwise | str}"
+
+# Ported from the jq tutorial
+export def example1 (j: JValue): Result JValue Error = Pass j
+export def example2 (j: JValue): Result JValue Error =
+    j | jAt 0 | expectOne ""
+
+# Simplify github json first commit
+export def example3 (log: JValue): Result JValue Error =
+    require Pass commit =
+        single log
+        | Pass
+        |> jAt 0
+        | expectOne "Head of log not found"
+
+    require Pass message =
+        single commit
+        | Pass
+        |> jField "commit"
+        |> jField "message"
+        | expectOne "commit message not found"
+
+    require Pass name =
+        single commit
+        | Pass
+        |> jField "commit"
+        |> jField "name"
+        | expectOne "commit author not found"
+
+    def output = JObject (
+        "message" :-> message,
+        "name" :-> name,
+        Nil
+    )
+
+    Pass output
+
+# Simplify full github json log
+export def example4 (log: JValue): Result JValue Error =
+    def simpleCommit commit =
+        require Pass message =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "message"
+            | expectOne ".commit.message missing"
+
+        require Pass name =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "name"
+            | expectOne ".commit.name missing"
+
+        def output = JObject (
+            "message" :-> message,
+            "name" :-> name,
+            Nil
+        )
+
+        Pass output
+
+    single log
+    | Pass
+    |> jFlatten
+    |: findFailFn simpleCommit
+    | rmap JArray
+
+# Simplify full github json log but keep parent urls
+export def example5 (log: JValue): Result JValue Error =
+    def simpleCommit commit =
+        require Pass message =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "message"
+            | expectOne ".commit.message missing"
+
+        require Pass name =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "name"
+            | expectOne ".commit.name missing"
+
+        require Pass parents =
+            single commit
+            | Pass
+            |> jField "parents"
+            |> jFlatten
+            |> jField "html_url"
+            |: makePass JArray
```

Comment:
Just leave off the `makePass JArray` step entirely, and below write `"parents" :-> JArray parents,`